File: frontend/src/api/endpoints/mechanics.ts
/** Client de /api/v1/mechanics. */
import api from "@/api/client";

export interface SkillRead {
  category: string;
  proficiency: number;
  certified: boolean;
}

export interface MechanicRead {
  id: string;
  user_id: string;
  email: string;
  branch_id: string | null;
  level: "junior" | "intermedio" | "master";
  capacity_hrs_day: number;
  current_load_hrs: number;
  available_hrs: number;
  load_status: "green" | "yellow" | "red";
  active_assignments_count: number;
  active: boolean;
  skills: SkillRead[];
}

export interface MechanicProfileCreatePayload {
  user_id: string;
  level?: "junior" | "intermedio" | "master";
  employee_number?: string;
  capacity_hrs_day?: number;
  hourly_cost?: number;
}

export interface MechanicProfileUpdatePayload {
  level?: "junior" | "intermedio" | "master";
  capacity_hrs_day?: number;
  hourly_cost?: number;
  active?: boolean;
  notes?: string;
}

export interface SkillUpsertPayload {
  category: string;
  proficiency: number;
  certified: boolean;
}

export const mechanicsApi = {
  list: (params: { only_active?: boolean; min_level?: string } = {}) => {
    const qs = new URLSearchParams(params as Record<string, string>).toString();
    return api.get<MechanicRead[]>(`/v1/mechanics${qs ? "?" + qs : ""}`).then((r) => r.data);
  },

  create: (payload: MechanicProfileCreatePayload) =>
    api.post<MechanicRead>(`/v1/mechanics`, payload).then((r) => r.data),

  update: (userId: string, payload: MechanicProfileUpdatePayload) =>
    api.patch<MechanicRead>(`/v1/mechanics/${userId}`, payload).then((r) => r.data),

  addSkill: (userId: string, payload: SkillUpsertPayload) =>
    api.post<SkillRead>(`/v1/mechanics/${userId}/skills`, payload).then((r) => r.data),
};

File: app/schemas/mechanics.py
"""Schemas para /api/v1/mechanics."""
from __future__ import annotations

from typing import Literal, Optional

from pydantic import BaseModel, ConfigDict, Field


LevelLiteral = Literal["junior", "intermedio", "master"]
SkillCategoryLiteral = Literal[
    "frenos", "motor", "transmision", "suspension", "electrico",
    "diagnostico", "hojalateria", "afinacion", "diesel", "otros",
]


class SkillRead(BaseModel):
    category: str
    proficiency: int = Field(..., ge=1, le=5)
    certified: bool

    model_config = ConfigDict(from_attributes=True)


class MechanicRead(BaseModel):
    id: str
    user_id: str
    email: str
    branch_id: Optional[str] = None
    level: LevelLiteral
    capacity_hrs_day: float
    current_load_hrs: float
    available_hrs: float
    load_status: Literal["green", "yellow", "red"]
    active_assignments_count: int
    active: bool
    skills: list[SkillRead]


class MechanicProfileCreate(BaseModel):
    user_id: str
    level: LevelLiteral = "junior"
    employee_number: Optional[str] = None
    capacity_hrs_day: float = 8.0
    hourly_cost: Optional[float] = None


class MechanicProfileUpdate(BaseModel):
    level: Optional[LevelLiteral] = None
    capacity_hrs_day: Optional[float] = None
    hourly_cost: Optional[float] = None
    active: Optional[bool] = None
    notes: Optional[str] = None


class SkillUpsert(BaseModel):
    category: SkillCategoryLiteral
    proficiency: int = Field(3, ge=1, le=5)
    certified: bool = False
